kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/ContainerOrchestrationServiceImpl.java

```diff
@@ -736,14 +736,16 @@ private boolean doesImageExist(String imageName, String imageTag) {
 
         List<Image> images = this.dockerClient.listImagesCmd().exec();
 
+        String requiredImage = imageName + ":" + imageTag;
+
         for (Image image : images) {
-            if (image.getRepoTags() != null) {
-                for (String tags : image.getRepoTags()) {
-                    if (tags.contains(imageName + ":" + imageTag)) {
-                        return true;
-                    }
+            if (isNull(image.getRepoTags())) {
+                continue;
+            }
+            for (String tag : image.getRepoTags()) {
+                if (tag.equalsIgnoreCase(requiredImage)) {
```

Comment:
looks good, my only concern is it appears that container tags can be case-sensitive.

- https://www.nocentino.com/posts/2019-09-03-docker-pull-is-case-sensitive/
- https://docs.docker.com/engine/reference/commandline/tag/

![image](https://user-images.githubusercontent.com/29900100/227966606-9efdf2e5-8ba4-4546-8174-656af9094428.png)

Reply:
ok, I'll change the equalsignorecase to a plain equals